src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -1133,7 +1119,14 @@ private unsafe SocketError FinishOperationConnect()
             }
         }
 
-        private unsafe int GetSocketAddressSize() => *(int*)PtrSocketAddressBufferSize;
+        private unsafe int GetSocketAddressSize()
+        {
+            Debug.Assert(SocketAddressNativeMemory != IntPtr.Zero);
+            int size = *((int*)SocketAddressNativeMemory);
+            _socketAddress!.Size = size;
+            new Span<byte>((void*)PtrSocketAddressBuffer(), size).CopyTo(_socketAddress.Buffer.Span);
```

Comment:
Why is it copying the buffer as part of getting the size?

Reply:
that was convenient hook to do that. I updated method name to properly outline current purpose.